pyiron_base/storage/has_stored_traits.py

```diff
@@ -6,17 +6,14 @@
 A parent class for managing input to pyiron jobs.
 """
 
-from __future__ import annotations
-
 from abc import ABC, ABCMeta
-from typing import Optional, TYPE_CHECKING
+from typing import Optional
 
 from traitlets import HasTraits, MetaHasTraits
 
 from pyiron_base.interfaces.object import HasStorage
 
-if TYPE_CHECKING:
-    from pyiron_base.storage.hdfio import ProjectHDFio
```

Comment:
So the `TYPE_CHECKING` is not strictly necessary filter is not strictly necessary here -- there are no circular imports -- but I like to use it anyhow to signal intent: this import is used _only_ for type hinting in this module, it's not actually called. (There is potentially some speedup too, since `TYPE_CHECKING` is just `False` and this way we avoid importing all the stuff inside the clause at run-time -- but that's not my motivation.) 

My opinion here is not strong but is completely unambiguous; i.e. it's not a hill I want to die on, but I'd rather keep this.

Reply:
If I keep the `TYPE_CHECKING` the tests fail - see first commit.

Reply:
I guess `annotations` is doing some work here after all. Then my preference would be to keep both `from __future__ import annotations` and `if TYPE_CHECKING`, which should be the previous-and-passing state, yes?

Reply:
`from __future__ import annotations` is [scheduled to become mandatory (automatic) behaviour eventually](https://docs.python.org/3/library/__future__.html#id1) anyhow